targets/targets.json

```diff
@@ -1276,7 +1276,7 @@
         "inherits": ["Target"],
         "core": "Cortex-M4F",
         "supported_toolchains": ["ARM", "uARM", "GCC_ARM", "IAR"],
-        "extra_labels": ["STM", "STM32F4", "STM32F411RE"],
+        "extra_labels": ["STM", "STM32F4", "STM32F411RE", "DRAGONFLY_MODEM"],
```

Comment:
Cellular is a feature. A single Cellular driver serves many different variants. However, a modem can be common in all of those variants. Mostly in mbed environment, the modem is driven by an external MCU.  TARGET_UBLOX_C027 or TARGET_MTS_DRAGONFLY_F411RE are sub-directories under NXP and STM families providing board specific initialization. It has nothing to do with the modem specific settings. We have tried to make PinNames uniform across all cellular platforms (will be done for any future porting) wo that a modem driver can handle any board configuration.  We are following here mbed conventions, as its STM family -> a specific STM class of MCUs -> a specific MCU -> attached to a specific modem. 
If for example you don't have this structure then, we need to have multiple copies of the same driver in different target defined directories which is a massive maintainance headache.